Read initial EMMS stream definitions from file

exot opened the pull request:
Upon first invocation, `helm-emms` reads the stream definitions from the builtin EMMS stream list, ignoring any user defined streams.  This is because the new EMMS streams implementation has introduced `emms-streams-built-in-list` as a non-customizable variable.  To accomodate this, `helm-emms` now first checks the content of `emms-streams-file` before falling back to the builtin streams list.

exot committed:
Read initial EMMS stream definitions from streams file

In EMMS 6.0, streams are defined in a native playlist located in
`emms-streams-file'.  When no streams are definied initially in
`helm-emms-streams-list', let `helm-emms' read the list from `emms-streams-file'
by default, and fall back to `emms-streams-built-in-list' if that is nil.  By
this, user defined streams are shown by `helm-emms' instead of the builtin ones.

exot committed:
Update package headers

Collect present information into recommended header values.

thierryvolpiatto commented:
Daniel Borchmann <notifications@github.com> writes:

> Upon first invocation, helm-emms reads the stream definitions from the
> builtin EMMS stream list, ignoring any user defined streams. This is
> because the new EMMS streams implementation has introduced
> emms-streams-built-in-list as a non-customizable variable.

So where your "user defined streams" come from? See below.

> To accomodate this, helm-emms now first checks the content of
> emms-streams-file before falling back to the builtin streams list.

This is unclear, until now AFAIU emms didn't allow configuring streams,
where are your "user defined streams"?

Note that the emms-streams-file contents is listing at first only
emms-streams-built-in-list (not configurable as you mention above),
helm-emms have a user var `helm-emms-streams-list` that allows
configuring streams easyly, what's the problem with it?

Otherwise don't mix unrelated things in your PR's, we have here some
checkdoc and headers fixes along with code changes which make difficult
reviewing, if you want to make such changes make them as separate
commits.

Thanks.


> ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/emacs-helm/helm-emms/pull/24
>
> Commit Summary
>
>   • Read initial EMMS stream definitions from streams file
>   • Update package dependencies and test local installation
>   • Fix some issues reported by checkdoc
>
> File Changes
>
>   • M helm-emms.el (46)
>
> Patch Links:
>
>   • https://github.com/emacs-helm/helm-emms/pull/24.patch
>   • https://github.com/emacs-helm/helm-emms/pull/24.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.*


-- 
Thierry

exot commented:
Hey Thierry,

If I remember correctly, `emms` used to have an interface to customize streams, via `emms-streams`.  The streams backend has been changed to store the streams in a native playlist now, which can be edited and exported directly from within emms.  I would propose that `helm-emms` takes this into account.   The variable `helm-emms-streams-list` is completely fine, but not updated by emms.  Indeed, I only learned about this variable when reading the code of `helm-emms`, I had missed it before.

However, I realized that my personal stream list did not have the right format, I will update this PR accordingly.

The update to the package metadata was necessary to reflect the new dependency on emms 6.0 (this should be version of EMMS introducing the streams rewrite), the checkdoc fixes are included to satisfy code quality requirements from melpa.  But I can take the latter into another PR.  I can also make the commits more fine-grained, no problem.

Best,

  Daniel

exot committed:
Test local installation via package-install-file

This is recommended by the contribution guide from melpa.  To make it work, the
final line had to be extended to include the `.el' suffix.

exot committed:
Fix bug when reading EMMS streams from file

`read-from-string' returns a cons, and we are only interested in the car of it.

thierryvolpiatto commented:
Now merged thanks.
However, next time could you make separate PRs each one relative to the same subject.
Thanks!

exot commented:
Will do, thanks!
